diff --git a/src/options.c b/src/options.c
index e63985e..9f11a25 100644
--- a/src/options.c
+++ b/src/options.c
@@ -149,7 +149,7 @@ void init_options(void) {
     opts.color_win_ansi = FALSE;
     opts.max_matches_per_file = 0;
     opts.max_search_depth = DEFAULT_MAX_SEARCH_DEPTH;
-#if defined(__APPLE__) || defined(__MACH__)
+#if defined(__APPLE__) || defined(__MACH__) || defined(__MVS__)
     /* mamp() is slower than normal read() on macos. default to off */
     opts.mmap = FALSE;
 #else
